src/style.css

```
@@ -0,0 +1,76 @@
+* {
+  margin: 0;
+}
+
+.blocks {
+  display: flex;
+  flex-wrap: wrap;
+  background: #000;
+}
+
+.blocks__block {
+  min-width: 300px;
+  height: 300px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  flex-basis: 300px;
+  flex-grow: 1;
+  background: red;
+  color: #fff;
+  font-size: 100px;
+  font-family: Arial, sans-serif;
+}
+
+.blocks__block:nth-child(2) {
+  background: rgba(255, 0, 0, 0.8);
+}
+
+.blocks__block:nth-child(3) {
+  background: rgba(255, 0, 0, 0.6);
+}
+
+.blocks__block:nth-child(4) {
+  background: rgba(255, 0, 0, 0.4);
+}
+
+.blocks__block:nth-child(5) {
+  background: rgba(255, 0, 0, 0.2);
+}
+
+.blocks__block:nth-child(6) {
+  background: rgb(0, 0, 0);
+}
+
+@media (min-width:600px) {
+  .blocks__block:nth-child(3) {
+    order: 1;
+  }
+
+  .blocks__block:nth-child(5),
+  .blocks__block:nth-child(6) {
+    order: 2;
+  }
+}
+
+@media (min-width:900px) {
+
+  .blocks__block:nth-child(3),
+  .blocks__block:nth-child(6) {
+    order: 0;
+  }
+
+  .blocks__block:nth-child(4) {
+    order: 2;
+  }
+
+  .blocks__block:nth-child(5) {
+    order: 1;
+  }
+}
+
+@media (min-width:1200px) {
+  .blocks__block {
+    flex-basis: calc(100% / 3);
```

```suggestion
    flex-basis: 33%;
```
Можна і так. Хоча твій варіант виглядає більш математично правильним

---

дякс)